finality-aleph/src/metrics.rs

```diff
@@ -71,11 +71,26 @@ pub(crate) enum Checkpoint {
 
 #[derive(Clone)]
 pub struct Metrics<H: Key> {
-    inner: Arc<Mutex<Inner<H>>>,
+    inner: Option<Arc<Mutex<Inner<H>>>>,
 }
 
 impl<H: Key> Metrics<H> {
-    pub fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+    pub fn register<F: FnOnce(PrometheusError)>(
+        registry: Option<&Registry>,
+        on_registration_failed: F,
+    ) -> Metrics<H> {
```

Comment:
Register fn with option looks strange to me. Just rename and change the signature of this fn to `new(registry: Option<_>) -> Result<_,_>` to get rid of the failure callback, its a more rust-like way of doing such thngs.

Reply:
Good idea with renaming to `new`. I agree that callbacks are ugly, but the thing is, ideally I'd like to return both `Error` and `Metrics` in case of a failure. I thought of replacing the signature to return `(Metrics<H>, Option<PrometheusError>)`, but finally I've decided to move error handling to that function instead.